Sources/BrowserServicesKit/Autofill/AutofillUserScript+SecureVault.swift

```diff
@@ -430,11 +430,13 @@ extension AutofillUserScript {
               let accountId = Int64(id) else {
             return
         }
+        let requestingDomain = hostForMessage(message)
 
         vaultDelegate?.autofillUserScript(self, didRequestCredentialsForAccount: Int64(accountId)) {
             guard let credential = $0,
                   let id = credential.account.id,
-                  let password = String(data: credential.password, encoding: .utf8) else { return }
+                  let password = String(data: credential.password, encoding: .utf8),
+                  credential.account.domain == requestingDomain else { return }
```

Comment:
@GioSensation, do you have any idea why don't send a response in case of failure here? Is it possible to send failure in _RequestVaultCredentialsForAccountResponse_?

Reply:
We don't have an error UI or flow. These cases should not be possible in normal usage because if we're here it means we already know credentials exist for this domain (we inject the keys/listeners only if credentials exist and show a credentials list in the tooltip picker). We technically have both `success` and `error` objects in the response but no platform has ever implemented anything with `error` in there, again because there is no UI.

What we should do here is probably send an error pixel because this shouldn't be possible.

Reply:
👍 , adjusted the unit test so there is not expectation for an answer in case of error

Reply:
Can we do 
```suggestion
                  credential.account.domain == requestingDomain else {
                      // Fire error pixel here if possible
                      replyHandler("{}")
                      return   
                  }
```